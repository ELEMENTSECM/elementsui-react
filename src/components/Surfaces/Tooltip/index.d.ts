import * as React from 'react';

export interface TooltipProps {
	/**
	 * Tooltip content
	 */
	content?: string;
}

export const Tooltip: React.SFC<TooltipProps>;
