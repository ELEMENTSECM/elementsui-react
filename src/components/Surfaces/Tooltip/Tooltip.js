import React from 'react';
import PropTypes from 'prop-types';
import { TooltipHost as UIFabTooltipHost } from 'office-ui-fabric-react/lib/Tooltip';

/**Tooltip example */
function Tooltip({ content, element }) {
	return (
		<div>
			<UIFabTooltipHost content={content}>{element}</UIFabTooltipHost>
		</div>
	);
}

Tooltip.propTypes = {
	/**Tooltip content */
	content: PropTypes.string
};

export default Tooltip;
