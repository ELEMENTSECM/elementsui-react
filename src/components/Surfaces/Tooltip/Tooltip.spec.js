import React from 'react';
import renderer from 'react-test-renderer';
import Tooltip from './Tooltip';

describe('Tooltip', () => {
	test('displays `Tooltip` as content', () => {
		const tree = renderer
			.create(<Tooltip htmlId="defaultTooltip" content="Tooltip" label="Foo" />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
