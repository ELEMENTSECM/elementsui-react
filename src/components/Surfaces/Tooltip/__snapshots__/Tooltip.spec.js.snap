// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip displays \`Tooltip\` as content 1`] = `
<div>
  <div
    aria-describedby={undefined}
    className="ms-TooltipHost host_a3f87333"
    onBlurCapture={[Function]}
    onFocusCapture={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  />
</div>
`;
